hydroflow/tests/datalog_frontend.rs

```diff
@@ -783,6 +783,30 @@ fn test_aggregations_and_comments() {
     assert_eq!(&res, &[(3, 1), (10, 3)]);
 }
 
+#[multiplatform_test]
+fn test_choose_strings() {
+    let (strings_send, strings) = hydroflow::util::unbounded_channel::<(String,)>();
+    let (result, mut result_recv) = hydroflow::util::unbounded_channel::<(String,)>();
+
+    let mut flow = datalog!(
+        r#"
+        .input strings `source_stream(strings)`
+        .output result `for_each(|v| result.send(v).unwrap())`
+
+        result(choose(a)) :- strings(a)
+        "#
+    );
+
+    strings_send.send(("hello".to_string(),)).unwrap();
+
+    flow.run_tick();
+
+    assert_eq!(
+        &collect_ready::<Vec<_>, _>(&mut result_recv),
+        &[("hello".to_string(),)]
+    );
+}
+
```

Comment:
Are they still called integers internally even tho these are strings?

Reply:
No this is the "fix aggregation lifetimes" part haha, probably should have created separate PRs but too late...

Reply:
Oops my fault for only using reviewstack on the other pr